### ISSUE TITLE
Photo issue

### ISSUE THREAD

**Issue by 40417256** [39aldo39/Evolution-DecSync#45]
Phone android 9, DecSyncCC2.1.0
PC Evolution 3.36.5 ubuntu, libdecsync-v2.0.1, evolution-decsync-v2.0.0-evolution-3.36

Creating or modifying a contact's information from the phone will result in photo being deleted in evolution (in case of modification existing photo will be deleted).

I also get these in my logs:

Nov  3 11:29:54 test1 evolution-addre[3607]: invalid unclassed pointer in cast to 'EBookBackendDecsync'
Nov  3 11:29:54 test1 evolution-addressbook-factory[3607]: DecSync: Execute all new entries

**Comment by 39aldo39** [comment]
I can reproduce the issue. The photo from Android is not recognized by Evolution. However, exporting the contact from Evolution still contains the photo, so it looks like a bug in Evolution itself, but I have to investigate.

**Comment by 40417256** [comment]
I have to disagree, if I sent the card by email and import it into Evolution the photo is there. The problem lies between DecSyncCC and Evolution-DecSync.

**Comment by 39aldo39** [comment]
I have investigated it more and it is a problem of Evolution, but DecSync CC can work around it.

DecSync CC currently writes a VCard version 4.0. Evolution does not support this correctly and (mainly) will not display the photo (see [this issue](https://gitlab.gnome.org/GNOME/evolution-data-server/-/issues/8)). Exporting a contact from the contacts application writes it as a VCard version 2.1, which is correctly handled. I have updated DecSync CC to write in version 3.0, working around the lacking support.
